server/pypi/make.sh

```diff
@@ -1,57 +1,285 @@
+
+eval "$(command conda 'shell.bash' 'hook')"
+conda activate base
+
+if [ "${BUILD_ENVS}" == "1" ]; then
+  rm -rf "${LOGS}" "${TOOLCHAINS}"
+  curl --silent --location "https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-macos-universal.tar.gz" --output cmake.tar.gz
+  mkdir -p "${TOOLCHAINS}"
+  tar -xzf cmake.tar.gz --directory "${TOOLCHAINS}"
+  mv "${TOOLCHAINS}/cmake-${CMAKE_VERSION}"*/CMake.app "${TOOLCHAINS}"
+  rm -rf "${TOOLCHAINS}/cmake-${CMAKE_VERSION}"* cmake.tar.gz
+
+  for PYTHON_VERSION in ${PYTHON_VERSIONS}; do
+    rm -rf "$(conda info | grep 'envs directories' | awk -F ':' '{ print $2 }' | sed -e 's/^[[:space:]]*//')/${CONDA_ENV:?}"
+    conda create -y --name "${CONDA_ENV}-${PYTHON_VERSION}" "python==${PYTHON_EXACT_VERSIONS[${PYTHON_VERSION}]}"
+    conda activate "${CONDA_ENV}-${PYTHON_VERSION}"
+    pip3 install -r requirements.txt
+    curl --silent --location "$(curl --silent -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" --location https://api.github.com/repos/beeware/Python-Apple-support/releases | grep browser_download_url | grep iOS | grep "${PYTHON_VERSION}" | head -1 | awk '{ print $2 }' | tr -d '"')" --output "python-apple-support-${PYTHON_VERSION}.tar.gz"
+    mkdir -p "${TOOLCHAINS}/${PYTHON_VERSION}" "${LOGS}/${PYTHON_VERSION}"
+    tar -xzf "python-apple-support-${PYTHON_VERSION}.tar.gz" --directory "${TOOLCHAINS}/${PYTHON_VERSION}"
+    rm -rf "python-apple-support-${PYTHON_VERSION}.tar.gz" "${LOGS:?}/${PYTHON_VERSION:?}"/*
+  done
+fi
+
+# build dependencies
+
+if [ "${BUILD_DEPS}" == "1" ]; then
+  mkdir -p "${LOGS}/deps"
+  pushd "${TOOLCHAINS}"
+
+  if ! [ -d "${PYTHON_APPLE_SUPPORT}" ]; then
+    git clone https://github.com/beeware/Python-Apple-support.git
+  fi
+
+  conda activate "${CONDA_ENV}-3.10"
+  pushd "${PYTHON_APPLE_SUPPORT}"
+  git checkout 3.10
+  make libFFI-wheels
+  # TODO: make OpenSSL-wheels
+  make BZip2-wheels
+  make XZ-wheels
+  popd
+  popd
+
+  mkdir -p "${DIST_DIR}"
+  rm -rf "${DIST_DIR}/bzip2" "${DIST_DIR}/libffi" "${DIST_DIR}/xz"
+  mv -f "${TOOLCHAINS}/${PYTHON_APPLE_SUPPORT}/wheels/dist"/* "${DIST_DIR}"
+
+  for DEPENDENCY in ${DEPENDENCIES}; do
+    printf "\n\n*** Building dependency %s ***\n\n" "${DEPENDENCY}"
+    python build-wheel.py --toolchain "${TOOLCHAINS}" --python 3.10 iOS "${DEPENDENCY}" 2>&1 | tee "${LOGS}/deps/${DEPENDENCY}.log"
+
+    # shellcheck disable=SC2010
+    if [ "$(ls "dist/${DEPENDENCY}" | grep -c py3)" -ge "2" ]; then
+      echo "${DEPENDENCY}" >> "${LOGS}/success.log"
+    else
+      echo "${DEPENDENCY}" >> "${LOGS}/fail.log"
+    fi
+  done
+fi
+
+# build packages
+
+rm -f "${LOGS}/success.log" "${LOGS}/fail.log"
+touch "${LOGS}/success.log" "${LOGS}/fail.log"
+
+for PACKAGE in ${PACKAGES}; do
+  PACKAGE_VERSIONS="$(./versions.py "${YEAR}" "${PACKAGE}")"
+  printf "\n### Attempting package %s with versions: %s ###\n" "${PACKAGE}" "${PACKAGE_VERSIONS}"
```

Comment:
This is a neat trick! 

Does this require any additional handling to account for patches that drift over time? I'd be surprised if the same iOS patches apply cleanly for every release over a multi-year window. What's the failure mode when this sort of drift occurs?

Reply:
It's a temp solution. My proposal in the PR text is to merge this with the `build-wheel.py`. We could then only build a default version or many like I do here but yes, to do a nice job those patches should be per version if a match is detected or fallback to a default patch. Honestly, so far very few patches failed. Most of the failures are with Python 3.11.

The patches failing are failing with the unmodified code so we need to look at them.